src/sidebar/components/ThreadList.tsx

```diff
@@ -211,12 +212,22 @@ export default function ThreadList({ threads }: ThreadListProps) {
     }
   }, [store, newAnnotationTag]);
 
-  const mostRecentlyHighlightedAnnotationId = useMemo(
-    // If multiple highlighted annotations exist, assume that the last one in
-    // the list is the most recent.
-    () => highlightedAnnotations[highlightedAnnotations.length - 1],
-    [highlightedAnnotations],
-  );
+  const mostRecentlyHighlightedAnnotationId = useMemo(() => {
+    const highlightedAnnos = allAnnotations.filter(
+      anno => anno.id && highlightedAnnotations.includes(anno.id),
+    );
```

Comment:
If this becomes problematic, we can always go back to the previous more naive but more efficient approach.

Reply:
I'm also thinking now that perhaps we could `reduce` instead of `sort`. That should be more efficient as it is a single iteration of the list.

Reply:
The theoretical issue here is that if you had eg. 1000 annotations and 1000 updates, you'd do a million operations. A fix would be to put the highlighted annotation IDs in a set first. This will be slower in the common case where the `highlightedAnnotations` list is short though.

As I said, I think this is a theoretical rather than practical concern. Having hundreds of annotations does sometimes happen, having 1000 updates at the same time is unlikely.

Reply:
Based on this example, `reduce` seems clearly faster.

```js
const numbers = Array.from({ length: 10_000_000 }, (_, i) =>
  `${i}`.padStart(7, '0'),
);

console.time('reducing');
const maxReducing = numbers.reduce((max, current) => {
  return max === null || current > max ? current : max;
}, null);
console.timeEnd('reducing');

console.time('sorting');
numbers.sort((a, b) => {
  if (a === b) {
    return 0;
  }

  return a < b ? 1 : -1;
});
console.timeEnd('sorting');

const maxSorting = numbers[0];

console.log({ maxReducing, maxSorting });
```

Output for me:

```
reducing: 85.934ms
sorting: 179.138ms
{ maxReducing: '9999999', maxSorting: '9999999' }
```

The amount is negligible, but I think the code is going to be easier to reason about, and it has the benefit of not mutating the array as `sort` does, so I think I'm going to change it.

Reply:
> The theoretical issue here is that if you had eg. 1000 annotations and 1000 updates, you'd do a million operations. A fix would be to put the highlighted annotation IDs in a set first. This will be slower in the common case where the highlightedAnnotations list is short though.

Oh, so you think the bottleneck would be in the annotations filtering. Sorry, I misunderstood.